arch/sim/src/sim/up_i2cbuslinux.c

```diff
@@ -0,0 +1,310 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+#include <sys/ioctl.h>
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syslog.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include <linux/i2c-dev.h>
+#include <linux/i2c.h>
+
+#include "up_i2cbus.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define ERROR(fmt, ...) \
+        syslog(LOG_ERR, "up_i2cbuslinux: " fmt "\n", ##__VA_ARGS__)
+#define INFO(fmt, ...) \
+        syslog(LOG_ERR, "up_i2cbuslinux: " fmt "\n", ##__VA_ARGS__)
+#define DEBUG(fmt, ...)
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct linux_i2cbus_master_s
+{
+  const struct i2c_ops_s *ops; /* I2C vtable */
+  int file;
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int linux_i2cbus_transfer(struct i2c_master_s *dev,
+                                 struct i2c_msg_s *msgs, int count);
+#ifdef CONFIG_I2C_RESET
+static int linux_i2cbus_reset(struct i2c_master_s *dev);
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct i2c_ops_s i2c_linux_ops =
+{
+  .transfer = linux_i2cbus_transfer,
+#ifdef CONFIG_I2C_RESET
+  .reset = linux_i2cbus_reset,
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: linux_i2cbus_reset
+ *
+ * Description:
+ *   Provide i2c reset
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_I2C_RESET
+static int linux_i2cbus_reset(struct i2c_master_s *dev)
+{
+  return -1; /* Not implemented */
```

Comment:
-NOSYS

Reply:
Do we guarantee that the error numbers map?  I thought we just had to throw a generic -1 across the boundary?

Reply:
Even the error number isn't exactly match between Linux and NuttX, it is always better to return the difference error code than -1. Eating the real error code just make the debugging more tough.